vote_on_sns_proposal.sh

```diff
@@ -4,25 +4,27 @@ set -euo pipefail
 
 cd -- "$(dirname -- "${BASH_SOURCE[0]}")"
 
-export NUM_PARTICIPANTS="${1:-3}"
-export PROPOSAL="${2:-3}"
+export NUM_PARTICIPANTS="${1:-100}"
+export PROPOSAL="${2:-2}"
 export VOTE="${3:-y}"
 
 . ./constants.sh normal
 
+SNS_GOVERNANCE_CANISTER_ID=$(jq -r '.governance_canister_id' sns_canister_ids.json)
+
 for (( c=0; c<${NUM_PARTICIPANTS}; c++ ))
 do
   export ID="$(printf "%03d" ${c})"
   export NEW_DX_IDENT="participant-${ID}"
   export PEM_FILE="$(readlink -f ~/.config/dfx/identity/${NEW_DX_IDENT}/identity.pem)"
   dfx identity use "${NEW_DX_IDENT}"
   export DX_PRINCIPAL="$(dfx identity get-principal)"
-  export JSON="$(dfx canister --network "${NETWORK}" call sns_governance list_neurons "(record {of_principal = opt principal\"${DX_PRINCIPAL}\"; limit = 0})" | idl2json | jq -r ".neurons")"
+  export JSON="$(dfx canister --network "${NETWORK}" call "${SNS_GOVERNANCE_CANISTER_ID}" --candid candid/sns_governance.did list_neurons "(record {of_principal = opt principal\"${DX_PRINCIPAL}\"; limit = 0})" | idl2json | jq -r ".neurons")"
   for((i=0; i<"$(echo $JSON | jq length)"; i++))
   do
     export NEURON_ID="$(echo $JSON | jq -r ".[${i}].id[0].id" | python3 -c "import sys; ints=sys.stdin.readlines(); sys.stdout.write(bytearray(eval(''.join(ints))).hex())")"
     quill sns --canister-ids-file ./sns_canister_ids.json --pem-file "${PEM_FILE}" register-vote --proposal-id ${PROPOSAL} --vote ${VOTE} ${NEURON_ID} > msg.json
-    quill --insecure-local-dev-mode send --yes msg.json
+    quill send --pem-file "${PEM_FILE}" --insecure-local-dev-mode --yes msg.json
```

Comment:
`--pem-file` should not be necessary here

Reply:
Without the PEM file, I think that the active DFX identity is the sender. Are you sure?

Reply:
The `send` commands expects a signed message that it merely sends over the network (the message could, e.g., be signed on a device without Internet connection and then sent from another device that does not have the private key and thus cannot sign):
```
Sends a signed message or a set of messages

USAGE:
    quill send [OPTIONS] [FILE_NAME]

ARGS:
    <FILE_NAME>    Path to the signed message (`-` for stdin)
```

Reply:
What you're saying makes sense, I just remembered observing that it didn't work as expected until I added those arguments. :-/

So I'll try removing these options and seeing if the CI pipeline still passes.

Reply:
Please try out my suggestion and let me know in case it doesn't work.